src/components/approval-type/ApprovalType.vue

```diff
@@ -117,148 +114,147 @@ export default class ApprovalType extends Vue {
 
-  /** Draft application date. */
+  /** The initial date the application for restoration was mailed to the company. */
   @Prop({ default: '' }) readonly applicationDate!: string
 
   /** Whether this section is invalid. */
   @Prop({ default: false }) readonly invalidSection!: boolean
 
-  /** Whether the display of court order section is a radio button. */
+  /** Whether to show the court order radio button. */
   @Prop({ default: true }) readonly isCourtOrderRadio!: boolean
 
+  /** Whether validation is enabled. */
+  @Prop({ default: false }) readonly validate!: boolean
+
   // Local properties
+  private approvalTypeSelected = null as ApprovalTypes
   private courtOrderNumberText = ''
-  private valid = false
-  private approvalTypeSelected = ''
   private noticeDateText = ''
   private applicationDateText = ''
 
   // For template
   readonly ApprovalTypes = ApprovalTypes
 
-  // Date Picker Rules
-  protected readonly datePickerRules = [(v: string) => !!v || 'Date is required']
+  /** Date picker rules. */
+  readonly datePickerRules = [(v: string) => !!v || 'Date is required']
 
-  // Text Field Rules
-  protected readonly courtOrderNumRules = [
+  /** Text field rules. */
+  readonly courtOrderNumRules = [
     (v: string) => (!v || !/^\s/g.test(v)) || 'Invalid spaces', // leading spaces
     (v: string) => (!v || !/\s$/g.test(v)) || 'Invalid spaces', // trailing spaces
-    (v: string) => (!v || !(v.length < 5)) || 'Court order number is invalid',
-    (v: string) => (!v || !(v.length > 20)) || 'Court order number is invalid',
+    (v: string) => (!v || !(v.length < 5)) || 'Court order number is too short',
+    (v: string) => (!v || !(v.length > 20)) || 'Court order number is too long',
     (v: string) => !!v || 'A Court Order number is required'
   ]
 
   /** Called when component is mounted. */
   mounted (): void {
-    // Copy props to mutable properties
+    // copy props to mutable properties
     if (this.approvedByRegistrar) {
-      this.courtOrderNumberText = ''
-      this.approvalTypeSelected = ApprovalTypes.VIA_REGISTRAR
-    } else if (this.courtOrderNumber) {
-      this.courtOrderNumberText = this.courtOrderNumber
-      this.approvalTypeSelected = ApprovalTypes.VIA_COURT_ORDER
-    } else {
-      // Default state (no button selected)
-      this.radioButtonChanged('')
+      // first set approval type, then set values
+      // otherwise, validation will fail
+      this.setApprovalTypeSelected(ApprovalTypes.VIA_REGISTRAR)
+      this.setNoticeDateText(this.noticeDate)
+      this.setApplicationDateText(this.applicationDate)
+    } else if (this.approvedByCourtOrder) {
+      // first set approval type, then set values
+      // otherwise, validation will fail
+      this.setApprovalTypeSelected(ApprovalTypes.VIA_COURT_ORDER)
+      this.setCourtOrderNumberText(this.courtOrderNumber)
     }
-    if (!this.isCourtOrderRadio) {
-      this.approvalTypeSelected = ApprovalTypes.VIA_COURT_ORDER
```

Comment:
There is no more need for "default" state handling. When we come in to this component, we should have either `approvedByRegistrar` or `approvedByCourtOrder` set.

If both are set (in error) then only the first one will have an effect.

If neither are set (in error) then no radio buttons are pre-selected, but the component should still function correctly once the user selects one.